src/main/java/org/opentripplanner/datastore/https/HttpsDataSourceMetadata.java

```diff
@@ -0,0 +1,83 @@
+package org.opentripplanner.datastore.https;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.http.Header;
+import org.apache.http.HttpHeaders;
+import org.apache.http.client.utils.DateUtils;
+
+/**
+ * HTTPS data source metadata returned by the HTTP server (HTTP headers).
+ */
+public class HttpsDataSourceMetadata {
+
+  static final String CONTENT_TYPE_APPLICATION_GZIP = "application/gzip";
+  static final String CONTENT_TYPE_APPLICATION_ZIP = "application/zip";
+
+  private static final Set<String> HTTP_HEADERS = Set.of(
+    HttpHeaders.CONTENT_TYPE,
+    HttpHeaders.CONTENT_LENGTH,
+    HttpHeaders.LAST_MODIFIED
+  );
+
+  private final String contentType;
+
+  private final long contentLength;
+  private final long lastModified;
+
+  public HttpsDataSourceMetadata(List<Header> headers) {
+    this(
+      headers
+        .stream()
+        .filter(header -> HTTP_HEADERS.contains(header.getName()))
+        .collect(Collectors.toUnmodifiableMap(Header::getName, Header::getValue))
+    );
+  }
+
+  public HttpsDataSourceMetadata(Map<String, String> headers) {
+    contentType = headers.get(HttpHeaders.CONTENT_TYPE);
+    contentLength = parseLong(headers.get(HttpHeaders.CONTENT_LENGTH));
+    lastModified = parseDate(headers.get(HttpHeaders.LAST_MODIFIED));
+  }
+
+  public String contentType() {
+    return contentType;
+  }
+
+  public long contentLength() {
+    return contentLength;
+  }
+
+  public long lastModified() {
+    return lastModified;
+  }
+
+  public boolean isZipContentType() {
+    return CONTENT_TYPE_APPLICATION_ZIP.equalsIgnoreCase(contentType());
+  }
+
+  public boolean isGzipContentType() {
+    return CONTENT_TYPE_APPLICATION_GZIP.equalsIgnoreCase(contentType());
+  }
+
+  private static long parseDate(String lastModifiedHeader) {
+    if (lastModifiedHeader != null) {
+      Date lastModifiedDate = DateUtils.parseDate(lastModifiedHeader);
+      if (lastModifiedDate != null) {
+        return lastModifiedDate.getTime();
+      }
+    }
+    return 0;
```

Comment:
It's very much a matter of personal taste but I dislike repeated null checks (an null checks in general).

How to you like this?
```suggestion
    return Optional
      .ofNullable(lastModifiedHeader)
      .map(DateUtils::parseDate)
      .map(Date::getTime)
      .orElse(0L);
```

Reply:
I have no strong opinion about this, but it seems @t2gran has :-)
https://github.com/opentripplanner/OpenTripPlanner/pull/4480#discussion_r984441515

Reply:
I would just hoe that there would be a clean syntax for this, similar to how [optional chaining in JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) is. I really dislike how the repeated null-checks look

Reply:
Maybe you could also chime in here: https://github.com/opentripplanner/OpenTripPlanner/pull/4480#discussion_r984441515

Reply:
The `Optional` is not made for this, and it reduces readability - I wish Java had support for a safe operator like Kotlin(`?.`) as well, but until then, use the language feature, not a library feature intended for something else. I would probably written the above code like this:
```Java
if (lastModifiedHeader == null) {
  return -1;
}
Date lastModifiedDate = DateUtils.parseDate(lastModifiedHeader);
if (lastModifiedDate != null) {
  return -1;
}
return lastModifiedDate.getTime();
```
PS ! The method should return `-1` not `0` if the value does not exist https://github.com/opentripplanner/OpenTripPlanner/blob/efe8374473b611c2aac59e9e861f64991fb55737/src/main/java/org/opentripplanner/datastore/api/DataSource.java#L69